packages/peregrine/lib/talons/CheckoutPage/ShippingInformation/AddressForm/useGuestForm.js

```diff
@@ -11,12 +11,36 @@ export const useGuestForm = props => {
         shippingData
     } = props;
 
+    const errorRef = useRef();
+
     const [{ cartId }] = useCartContext();
 
-    const [setGuestShipping, { loading }] = useMutation(
+    const [setGuestShipping, { error, loading }] = useMutation(
         setGuestShippingMutation
     );
 
+    const derivedErrorMessage = useMemo(() => {
+        let errorMessage;
+
+        if (error) {
+            const { graphQLErrors, message } = error;
+            errorMessage = graphQLErrors
+                ? graphQLErrors.map(({ message }) => message).join(', ')
+                : message;
+        }
+
+        return errorMessage;
+    }, [error]);
+
+    useEffect(() => {
+        if (error) {
+            errorRef.current.scrollIntoView({
```

Comment:
Does this sort of logic belong in the talon? It's presentational which was my concern.